open-metadata-implementation/adapters/open-connectors/repository-services-connectors/open-metadata-collection-store-connectors/graph-repository-connector/src/main/java/org/odpi/openmetadata/adapters/repositoryservices/graphrepository/repositoryconnector/GraphOMRSRelationshipMapper.java

```diff
@@ -36,9 +38,13 @@ class GraphOMRSRelationshipMapper {
 
     private static final Logger log = LoggerFactory.getLogger(GraphOMRSRelationshipMapper.class);
 
-    private String               repositoryName;
-    private String               metadataCollectionId;
-    private OMRSRepositoryHelper repositoryHelper;
+    private static final ObjectMapper OBJECT_MAPPER = new ObjectMapper();
+    private static final ObjectReader OBJECT_READER = OBJECT_MAPPER.reader();
```

Comment:
I have not looked in detail, but can this class be run multithreaded? If so it seems that having these objects as static might be problematic as different threads could access the same objects and pick up unexpected content from the other thread.

Reply:
Yes where possible we should use more specific reader/writer object as they are thread-safe by design

`... Instances are initially constructed by [ObjectMapper](https://fasterxml.github.io/jackson-databind/javadoc/2.5/com/fasterxml/jackson/databind/ObjectMapper.html) and can be reused, shared, cached; both because of thread-safety and because instances are relatively light-weight.`

from the java docs https://fasterxml.github.io/jackson-databind/javadoc/2.5/com/fasterxml/jackson/databind/ObjectReader.html

Reply:
looks good.